### ISSUE TITLE
WebPopupMenu offset

### ISSUE THREAD

**Issue by iMichka** [mgarin/weblaf#292]
Hi. When a stand-alone popup menu (JPopupMenu) is displayed with a right click, there is an offset in the X position. The offset is emphasized by the offset of the triangle. I think the tip of the triangle should be aligned with the getX() and getY() positions given by the MouseEvent, if possible.

![screenshot](https://cloud.githubusercontent.com/assets/3406519/8085325/d77f1a9a-0f8f-11e5-8d4b-cb1dad1ad225.png)

Is there maybe a quick solution for this problem ? Maybe a new ```PopupMenuWay``` should be introduced for this case in the ```WebPopupMenuPainter```.

**Comment by mgarin** [comment]
I didn't change default method to show popup menu so that you could have the raw position, but on the second thought it is kinda useless since you will always want to position the corner under the mouse and will have to calculate it manually.

I will probably include changes for this behavior in `styling` or a bit later after I finish it. I will post an update here as soon as changes are available.

For now I'm afraid there is no good solution - only manually place menu where it should be.

**Comment by iMichka** [comment]
Yes I think the raw position which is used here is the position around the transparent box which is used for the popup menu. Manually recalculating it on WebLaf's side is indeed needed.

I'll add a manual offset for the moment to fix it on my side. Thanks for the help.
I'll be happy to test the ```styling``` branch as soon as you think it is ready.

**Comment by mgarin** [comment]
I have almost finished a huge chunk of changes, it will be out soon. That will probably be the last big and important change I have to make. After that I will polish and fix some skins and components and it should be ready for some open testing.

**Comment by mgarin** [comment]
I will be reworking popup menu styling and popups in general in one of the next smaller updates.

I plan to add a better default style along with few other styles like the dropdown one it has right now, but instead of having hardcoded styles popup menu will be fully customizable. I have already made most necessary improvements in v1.2.9 update and just need to work on the components.
